### PR TITLE
Dovecot plugin for Full Text Search (FTS) with Xapian

### PR DESCRIPTION
**Pull request by ju1m**
###### Motivation for this change
Run server-side Xapian-powered Full Text Search (FTS) through IMAP into mails managed by Dovecot2.

###### Things done
Add fts-xapian plugin, loadable with `services.dovecot2.modules = [ pkgs.dovecot_fts_xapian ];`
Pandoc support is disabled.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by symphorien** [comment]
I tested it and it works :)

File: pkgs/servers/mail/dovecot/plugins/fts_xapian/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  '';
+
+  configureFlags = [
+    "--with-dovecot=${dovecot}/lib/dovecot"
+    "--without-dovecot-install-dirs"
+    "--with-moduledir=$(out)/lib/dovecot"
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/grosjo/fts-xapian;
```
Comment (mweinelt) [inline pkgs/servers/mail/dovecot/plugins/fts_xapian/default.nix:27]: URLs should be quoted per RFC45.

https://github.com/NixOS/rfcs/blob/master/rfcs/0045-deprecate-url-syntax.md

File: pkgs/servers/mail/dovecot/plugins/fts_xapian/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchFromGitHub, autoconf, automake, pkg-config, dovecot, libtool, xapian, icu64 }:
+stdenv.mkDerivation {
+  pname = "fts-xapian";
+  version = "1.2.11a";
```
Comment (mweinelt) [inline pkgs/servers/mail/dovecot/plugins/fts_xapian/default.nix:4]: Maybe update to https://github.com/grosjo/fts-xapian/releases/tag/1.3.

Comment (symphorien) [reply pkgs/servers/mail/dovecot/plugins/fts_xapian/default.nix:4]: 1.3.1 got out today, even.

File: pkgs/servers/mail/dovecot/plugins/fts_xapian/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchFromGitHub {
+    owner = "grosjo";
+    repo = "fts-xapian";
+    rev = "1.2.11a";
+    sha256 = "1n9bvv2rgnmmpm472y067yfzwah39ig1ziwyk5gmrbg0k54x81dk";
+  };
+
+  buildInputs = [ dovecot autoconf automake libtool pkg-config xapian icu64 ];
```
Comment (symphorien) [inline pkgs/servers/mail/dovecot/plugins/fts_xapian/default.nix:13]: autoconf, automake, libtool and pkg-config should be nativeBuildInputs

**Comment by ju1m** [comment]
Thanks @symphorien and @mweinelt. I've amended the commit to include your corrections. I've also added sqlite as a dependency because it is now required by upstream.

File: pkgs/servers/mail/dovecot/plugins/fts_xapian/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "--with-dovecot=${dovecot}/lib/dovecot"
+    "--without-dovecot-install-dirs"
+    "--with-moduledir=$(out)/lib/dovecot"
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/grosjo/fts-xapian";
+    description = "Dovecot FTS plugin based on Xapian";
+    license = licenses.lgpl21;
+    maintainers = with maintainers; [ ];
```
Comment (mweinelt) [inline pkgs/servers/mail/dovecot/plugins/fts_xapian/default.nix:32]: Please add yourself as maintainer.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 78780` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- dovecot_fts_xapian
</details>

**Review by mweinelt — APPROVED** [review]

**Comment by Ekleog** [comment]
Thank you! :)

**Pull request merged**